client-state/models/FormField.ts

```diff
@@ -0,0 +1,74 @@
+import { types, flow, getParentOfType } from 'mobx-state-tree'
+import { FieldKey } from '../../utils/api/definitions/fields'
+import { Form } from './Form'
+
+export const KeyValue = types.model({
+  key: types.string,
+  text: types.string,
+})
+
+const Category = KeyValue.named('Category')
+const Options = KeyValue.named('Options')
+const Default = KeyValue.named('Default')
+
+export const FormField = types
+  .model({
+    key: types.string,
+    type: types.string,
+    label: types.string,
+    category: Category,
+    order: types.number,
+    placeholder: types.maybe(types.string),
+    default: types.maybe(Default),
+    value: types.maybeNull(types.union(types.string, KeyValue)),
+    options: types.optional(types.array(Options), []),
+    error: types.maybe(types.string),
+  })
+  .views((self) => ({
+    get filled() {
+      return (
+        self.value !== undefined && self.value !== null && self.value !== ''
+      )
+    },
+    get hasError() {
+      return self.error !== undefined
+    },
+  }))
+  .actions((self) => ({
+    setValue(value: string) {
+      self.value = value
+    },
+    setError(error: string) {
+      self.error = error
+    },
+    clearValue() {
+      self.value = null
+    },
+    sanitizeInput() {
+      let val = ''
+
+      if (KeyValue.is(self.value)) {
+        if (self.key == FieldKey.LIVING_COUNTRY || FieldKey.MARITAL_STATUS) {
```

Comment:
Yeah legal status being a radio means I can just use the value provided but the other two need to be massaged into the format required by the API before sending them, but this might be a moot discussion because I don't think I need those anyway! 

The `KeyValue.is()` check should give me all I need and the API will always want the key, so I can remove that if entirely! Good eye!

Reply:
fixed with

```js
if (KeyValue.is(self.value)) {
  val = self.value.key
} else {
  if (self.value.includes('$')) {
    val = self.value.toString().replace('$', '').replace(',', '')
  } else {
    val = self.value.toString()
  }
}
```